server/server.go

```diff
@@ -180,6 +183,48 @@ func (server *HTTPServer) readReportForCluster(writer http.ResponseWriter, reque
 	}
 }
 
+// readReportForCluster method retrieves metainformations for report stored in
+// database and return the retrieved info to requester via response payload.
+// The payload has type types.ReportResponseMetainfo
+func (server *HTTPServer) readReportMetainfoForCluster(writer http.ResponseWriter, request *http.Request) {
+	clusterName, successful := readClusterName(writer, request)
+	if !successful {
+		// everything has been handled already
+		return
+	}
+
+	orgID, successful := readOrgID(writer, request)
+	if !successful {
+		return
+	}
+
+	reports, lastChecked, storedAt, err := server.Storage.ReadReportForCluster(orgID, clusterName)
+	if err != nil {
+		log.Error().Err(err).Msg("Unable to read report for cluster")
+		handleServerError(writer, err)
+		return
+	}
+
+	hitRulesCount := getHitRulesCount(reports)
+
+	// -1 as count in response means there are no rules for this cluster
+	// as opposed to no rules hit for the cluster
+	if hitRulesCount == 0 {
```

Comment:
This is already handled in the `getHitRulesCount` function. Isn't it?

Reply:
ah sure, lemme check :)

Reply:
fixed